lib/ui5Framework/npm/Registry.js

```diff
@@ -49,13 +49,33 @@ class Registry {
 	}
 	async _getPacoteOptions() {
 		if (!this._npmConfig) {
-			const {default: libnpmconfig} = await import("libnpmconfig");
-			const opts = {
-				cache: this._cacheDir,
-			};
-			const config = libnpmconfig.read(opts, {
-				cwd: this._cwd
-			}).toJSON();
+			const {default: Config} = await import("@npmcli/config");
+			const {typeDefs} = Config;
+			const configuration = new Config({
+				// Merge collected data. Done in impure manner- need to alter the
+				// "output" param
```

Comment:
The reasons to use flatten are several, but the main was that it's the only way I saw to provide a JSON with the data.

Altough, it's not included into the docs, I don't seem to find a reasonable internal usage of it. It's a param in the constructor and then used only to flatten the data, but I didn't find internal usage, despite it was not publicly documented.
Impurity means that we need to alter the second parameter, `output`, so when we call `.flat()`, it would return the desired output. Please take a look here: https://github.com/npm/cli/blob/latest/workspaces/config/lib/index.js#L232

Reply:
The [second option](https://github.com/ragestudio/corenode/blob/85ce600a9d22012a1323413fa2e4da17e903463a/packages/pkg-manager/src/lib/npmPublishLib/index.js#LL16C96-L16C96) would be available with the next release of the `config`. So, far for `6.2.0.` those are not available. 
It seems that they moved it and now got them back, especially the `flatten`.

So, this would be blocked until a new release is made.